node/pod.go

```diff
@@ -344,7 +345,10 @@ func (pc *PodController) deletePodHandler(ctx context.Context, key string) (retE
 	ctx, span := trace.StartSpan(ctx, "processDeletionReconcilationWorkItem")
 	defer span.End()
 
-	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	idx := strings.LastIndex(key, "/")
```

Comment:
I'm reluctant to mess with this in any way. I yield to @cpuguy83 if you want to pass messages through the queue like this.

Reply:
That said, the Pod UID must come from somewhere and I'm not sure where because:
- `knownPods` shouldn't have the _old_ Pod around
- there's no way to access the provider's `deletedPods` map

So, ignore my last comment.